apis/keda/v1alpha1/scaledobject_webhook.go

```diff
@@ -588,3 +594,7 @@ func getHpaName(so ScaledObject) string {
 
 	return so.Spec.Advanced.HorizontalPodAutoscalerConfig.Name
 }
+
+func getFunctionName(i interface{}) string {
+	return runtime.FuncForPC(reflect.ValueOf(i).Pointer()).Name()
```

Comment:
alternatively, if we go forward with `FuncForPC`, the code should probably check for `nil` ptr because `FuncForPC` can [return nil](https://cs.opensource.google/go/go/+/refs/tags/go1.23.5:src/runtime/symtab.go;l=735), [docs](https://pkg.go.dev/runtime#FuncForPC). This will future-proof the code a bit in case of go compiler optimizations changing this behavior